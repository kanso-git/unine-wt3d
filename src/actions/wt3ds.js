
// ADD_WT3D
export const addWt3d = (expense) => ({
  type: 'ADD_WT3D',
  expense
});

export const startAddWt3d = (expenseData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };

    return database.ref(`users/${uid}/wt3ds`).push(expense).then((ref) => {
      dispatch(addWt3d({
        id: ref.key,
        ...expense
      }));
    });
  };
};

// REMOVE_WT3D
export const removeWt3d = ({ id } = {}) => ({
  type: 'REMOVE_WT3D',
  id
});

export const startRemoveWt3d = ({ id } = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds/${id}`).remove().then(() => {
      dispatch(removeWt3d({ id }));
    });
  };
};

// EDIT_WT3D
export const editWt3d = (id, updates) => ({
  type: 'EDIT_WT3D',
  id,
  updates
});

export const startEditWt3d = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds/${id}`).update(updates).then(() => {
      dispatch(editWt3d(id, updates));
    });
  };
};

// SET_WT3DS
export const setWt3ds = (wt3ds) => ({
  type: 'SET_WT3DS',
  wt3ds
});

export const startSetWt3ds = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds`).once('value').then((snapshot) => {
      const wt3ds = [];

      snapshot.forEach((childSnapshot) => {
        wt3ds.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setWt3ds(wt3ds));
    });
  };
};
