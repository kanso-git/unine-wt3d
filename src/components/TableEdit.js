import React from 'react';
import 'react-table/react-table.css';
import ReactTable from 'react-table';
import { makeData, Logo, Tips } from './Utils';

class TableEdit extends React.Component {
  constructor() {
    super();
    this.state = {
      data: makeData(),
    };
  }
  render() {
    const { data } = this.state;
    return (
      <div>
        <ReactTable
          data={data}
          columns={[{
      Header: 'SecrÃ©tariats',
      columns: [{
        Header: 'Local Code',
        accessor: 'firstName',
      }, {
        Header: 'Local Name',
        id: 'lastName',
        accessor: d => d.lastName,
      }, {
        Header: '',
        accessor: 'firstName',
        Cell: row => (
          <div style={{ display: 'flex', flexDirection: 'row', justifyContent: 'center' }}>
            <button className="button button--delete" onClick={() => alert(row)} >
              delete
            </button>
          </div>
        ),
      }],
    }]}
          defaultPageSize={10}
          className="-striped -highlight"
        />
        <br />
        <Tips />
      </div>
    );
  }
}
export default TableEdit;
