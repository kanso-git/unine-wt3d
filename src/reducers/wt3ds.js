// wt3ds Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_WT3D':
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_WT3D':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_WT3D':
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        };
      });
    case 'SET_WT3DS':
      return action.wt3ds;
    default:
      return state;
  }
};
