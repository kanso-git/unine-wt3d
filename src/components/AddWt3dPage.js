import React from 'react';
import { connect } from 'react-redux';
import Wt3dForm from './Wt3dForm';
import { startAddWt3d } from '../actions/wt3ds';


export class AddWt3dPage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddWt3d(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add new waytouch</h1>
          </div>
        </div>
        <div className="content-container">
          <Wt3dForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddWt3d: expense => dispatch(startAddWt3d(expense)),

});


export default connect(undefined, mapDispatchToProps)(AddWt3dPage);
