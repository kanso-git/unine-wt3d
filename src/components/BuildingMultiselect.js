import React, { Component } from 'react';
import { connect } from 'react-redux';
import 'react-select/dist/react-select.css';
import Select from 'react-select';

import { bdl } from '../actions';

const staticdata = [
  { label: 'Chocolate', value: 'chocolate' },
  { label: 'Vanilla', value: 'vanilla' },
  { label: 'Strawberry', value: 'strawberry' },
  { label: 'Caramel', value: 'caramel' },
  { label: 'Cookies and Cream', value: 'cookiescream' },
  { label: 'Peppermint', value: 'peppermint' },
];

const WHY_WOULD_YOU = [
  { label: 'Chocolate (are you crazy?)', value: 'chocolate', disabled: true },
].concat(staticdata.slice(1));

class BuildingMultiselect extends Component {

  state={
    removeSelected: true,
    disabled: false,
    crazy: false,
    stayOpen: false || this.props.stayOpen,
    value: [],
    rtl: false,
  }
  componentDidMount() {
    this.getBuildingList();
  }
  getBuildingList = async () => {
    await this.props.getBuildingList();
  }
  handleSelectChange= (value) => {
    console.log('You\'ve selected:', value);
    this.setState({ value });
    this.props.handleSelect(value);
  }

  toggleCheckbox = (e) => {
    this.setState({
      [e.target.name]: e.target.checked,
    });
  }

  toggleRtl= (e) => {
    const rtl = e.target.checked;
    this.setState({ rtl });
  }

  render() {
    const {
      crazy, disabled, stayOpen, value,
    } = this.state;
    const options = crazy ? WHY_WOULD_YOU : this.props.buildingList;
    return (
      <div className="section">
        <h3 className="section-heading">{this.props.label} </h3>
        <Select
          closeOnSelect={!stayOpen}
          disabled={disabled}
          multi
          onChange={this.handleSelectChange}
          options={options}
          placeholder="Select your building(s)"
          removeSelected={this.state.removeSelected}
          rtl={this.state.rtl}
          simpleValue
          value={value}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const buildingList = state.bdl.buildingList;
  return { buildingList };
};

export default connect(mapStateToProps,{...bdl})(BuildingMultiselect);
