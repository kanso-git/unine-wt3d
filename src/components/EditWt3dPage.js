import React from 'react';
import { connect } from 'react-redux';
import Wt3dForm from './Wt3dForm';
import { startEditWt3d, startRemoveWt3d } from '../actions/wt3ds';

export class EditWt3dPage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditWt3d(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveWt3d({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Wt3d</h1>
          </div>
        </div>
        <div className="content-container">
          <Wt3dForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Wt3d</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.wt3ds.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditWt3d: (id, expense) => dispatch(startEditWt3d(id, expense)),
  startRemoveWt3d: (data) => dispatch(startRemoveWt3d(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditWt3dPage);
