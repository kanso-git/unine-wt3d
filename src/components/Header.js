import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faPowerOff } from '@fortawesome/fontawesome-free-solid';

import { startLogout } from '../actions/auth';

export const Header = ({ startLogout, auth }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Waytouch 3D</h1>
        </Link>
        <button className="button button--link" onClick={startLogout}>
          <FontAwesomeIcon icon={faPowerOff} /> Logout ({auth.user.displayName})
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
});

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
