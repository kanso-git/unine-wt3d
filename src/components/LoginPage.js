import React, { Component } from 'react';
import { connect } from 'react-redux';
import { authActions } from '../actions';

export class LoginPage extends Component {
  state={
    error: null,
    username: '',
    password: '',
  }
  
  onUsernameChange = (e) => {
    const username = e.target.value;
    this.setState(() => ({ username }));
  }
  onPasswordChange = (e) => {
    const password = e.target.value;
    this.setState(() => ({ password }));
  }
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.username || !this.state.password) {
      this.setState(() => ({ error: 'Please provide username and password.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.startLogin(this.state.username, this.state.password);
      e.target.reset();
    }
  };

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">Waytouch 3D</h1>
          <form className="form" onSubmit={this.onSubmit} >
            {this.state.error && <p className="form__error">{this.state.error}</p>}
            <input
              type="text"
              placeholder="Login"
              autoComplete="off"
              className="text-input"
              value={this.state.username}
              onChange={this.onUsernameChange}
            />
            <input
              type="password"
              placeholder="Password"
              className="text-input"
              autoComplete="off"
              value={this.state.password}
              onChange={this.onPasswordChange}
            />

            <p> It's time to get your Waytouch under control.</p>
            <button className="button" onClick={this.login} >Login</button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    auth: state.auth,
  });


export default connect(mapStateToProps, { ...authActions })(LoginPage);
