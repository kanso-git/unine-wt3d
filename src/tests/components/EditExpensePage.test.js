import React from 'react';
import { shallow } from 'enzyme';
import wt3ds from '../fixtures/wt3ds';
import { EditWt3dPage } from '../../components/EditWt3dPage';

let startEditWt3d, startRemoveWt3d, history, wrapper;

beforeEach(() => {
  startEditWt3d = jest.fn();
  startRemoveWt3d = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditWt3dPage
      startEditWt3d={startEditWt3d}
      startRemoveWt3d={startRemoveWt3d}
      history={history}
      expense={wt3ds[2]}
    />
  );
});

test('should render EditWt3dPage', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle startEditWt3d', () => {
  wrapper.find('Wt3dForm').prop('onSubmit')(wt3ds[2]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startEditWt3d).toHaveBeenLastCalledWith(wt3ds[2].id, wt3ds[2]);
});

test('should handle startRemoveWt3d', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startRemoveWt3d).toHaveBeenLastCalledWith({
    id: wt3ds[2].id
  });
});
