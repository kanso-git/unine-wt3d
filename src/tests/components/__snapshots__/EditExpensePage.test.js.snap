// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditWt3dPage 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Wt3d
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <Wt3dForm
      expense={
        Object {
          "amount": 4500,
          "createdAt": 345600000,
          "description": "Credit Card",
          "id": "3",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      Remove Wt3d
    </button>
  </div>
</div>
`;
